### ISSUE TITLE
Design problem - display the description of the box 

### ISSUE THREAD

**Issue by mohammadHmedat** [moloch--/RootTheBox#186]
There is a problem in the display the description of the box when enabling "Autoformat Description" it breaks the statements!!!

I sent an email for the issue with some pics to moloch38@gmail.com

Thanks

**Comment by jeffg2k** [comment]
@mohammadHmedat I pushed a change to the autoformat.  Can you verify it resolves your issue?

**Comment by mohammadHmedat** [comment]
/admin/view/game_objects - 404
/admin/view/game_levels - 404
/admin/upgrades/source_code_market - 404
http://x.x.x.x/user - 404
http://x.x.x.x/user/missions - 404
http://x.x.x.x/teams - 404

**Comment by mohammadHmedat** [comment]
[E 181115 21:43:07 web:1670] Uncaught exception GET /admin/view/game_objects (x.x.x.x)
    HTTPServerRequest(protocol='http', host='x.x.x.x', method='GET', uri='/admin/view/game_objects', version='HTTP/1.1', remote_ip='x.x.x.x')
    Traceback (most recent call last):
      File "/usr/local/lib/python2.7/dist-packages/tornado/web.py", line 1590, in _execute
        result = method(*self.path_args, **self.path_kwargs)
      File "/home/ubuntu/RootTheBox/libs/SecurityDecorators.py", line 63, in wrapper
        return method(self, *args, **kwargs)
      File "/home/ubuntu/RootTheBox/libs/SecurityDecorators.py", line 38, in wrapper
        return method(self, *args, **kwargs)
      File "/home/ubuntu/RootTheBox/libs/SecurityDecorators.py", line 93, in wrapper
        return method(self, *args, **kwargs)
      File "/home/ubuntu/RootTheBox/handlers/AdminHandlers/AdminGameObjectHandlers.py", line 362, in get
        self.render(uri[args[0]], errors=None)
      File "/usr/local/lib/python2.7/dist-packages/tornado/web.py", line 766, in render
        html = self.render_string(template_name, **kwargs)
      File "/usr/local/lib/python2.7/dist-packages/tornado/web.py", line 907, in render_string
        return t.generate(**namespace)
      File "/usr/local/lib/python2.7/dist-packages/tornado/template.py", line 346, in generate
        return execute()
      File "admin/view/game_objects_html.generated.py", line 160, in _tt_execute
        for box in Box.all():  # admin/view/game_objects.html:263 (via main.html:27)
      File "/home/ubuntu/RootTheBox/models/Box.py", line 109, in all
        return dbsession.query(cls).all()
      File "/usr/local/lib/python2.7/dist-packages/sqlalchemy/orm/query.py", line 2843, in all
        return list(self)
      File "/usr/local/lib/python2.7/dist-packages/sqlalchemy/orm/query.py", line 2995, in __iter__
        return self._execute_and_instances(context)
      File "/usr/local/lib/python2.7/dist-packages/sqlalchemy/orm/query.py", line 3018, in _execute_and_instances
        result = conn.execute(querycontext.statement, self._params)
      File "/usr/local/lib/python2.7/dist-packages/sqlalchemy/engine/base.py", line 948, in execute
        return meth(self, multiparams, params)
      File "/usr/local/lib/python2.7/dist-packages/sqlalchemy/sql/elements.py", line 269, in _execute_on_connection
        return connection._execute_clauseelement(self, multiparams, params)
      File "/usr/local/lib/python2.7/dist-packages/sqlalchemy/engine/base.py", line 1060, in _execute_clauseelement
        compiled_sql, distilled_params
      File "/usr/local/lib/python2.7/dist-packages/sqlalchemy/engine/base.py", line 1200, in _execute_context
        context)
      File "/usr/local/lib/python2.7/dist-packages/sqlalchemy/engine/base.py", line 1413, in _handle_dbapi_exception
        exc_info
      File "/usr/local/lib/python2.7/dist-packages/sqlalchemy/util/compat.py", line 265, in raise_from_cause
        reraise(type(exception), exception, tb=exc_tb, cause=cause)
      File "/usr/local/lib/python2.7/dist-packages/sqlalchemy/engine/base.py", line 1193, in _execute_context
        context)
      File "/usr/local/lib/python2.7/dist-packages/sqlalchemy/engine/default.py", line 509, in do_execute
        cursor.execute(statement, parameters)
      File "/usr/local/lib/python2.7/dist-packages/MySQLdb/cursors.py", line 205, in execute
        self.errorhandler(self, exc, value)
      File "/usr/local/lib/python2.7/dist-packages/MySQLdb/connections.py", line 36, in defaulterrorhandler
        raise errorclass, errorvalue
    OperationalError: (_mysql_exceptions.OperationalError) (1054, "Unknown column 'box.flag_submission_type' in 'field list'") [SQL: u'SELECT box.id AS box_id, box.created AS box_created, box.uuid AS box_uuid, box.corporation_id AS box_corporation_id, box.category_id AS box_category_id, box._name AS box__name, box._operating_system AS box__operating_system, box._description AS box__description, box._difficulty AS box__difficulty, box.game_level_id AS box_game_level_id, box._avatar AS box__avatar, box.autoformat AS box_autoformat, box.garbage AS box_garbage, box.flag_submission_type AS box_flag_submission_type \nFROM box'] (Background on this error at: http://sqlalche.me/e/e3q8)
[E 181115 21:43:07 BaseHandlers:192] Request from x.x.x.x resulted in an error code 500:
    Traceback (most recent call last):
      File "/usr/local/lib/python2.7/dist-packages/tornado/web.py", line 1590, in _execute
        result = method(*self.path_args, **self.path_kwargs)
      File "/home/ubuntu/RootTheBox/libs/SecurityDecorators.py", line 63, in wrapper
        return method(self, *args, **kwargs)
      File "/home/ubuntu/RootTheBox/libs/SecurityDecorators.py", line 38, in wrapper
        return method(self, *args, **kwargs)
      File "/home/ubuntu/RootTheBox/libs/SecurityDecorators.py", line 93, in wrapper
        return method(self, *args, **kwargs)
      File "/home/ubuntu/RootTheBox/handlers/AdminHandlers/AdminGameObjectHandlers.py", line 362, in get
        self.render(uri[args[0]], errors=None)
      File "/usr/local/lib/python2.7/dist-packages/tornado/web.py", line 766, in render
        html = self.render_string(template_name, **kwargs)
      File "/usr/local/lib/python2.7/dist-packages/tornado/web.py", line 907, in render_string
        return t.generate(**namespace)
      File "/usr/local/lib/python2.7/dist-packages/tornado/template.py", line 346, in generate
        return execute()
      File "admin/view/game_objects_html.generated.py", line 160, in _tt_execute
        for box in Box.all():  # admin/view/game_objects.html:263 (via main.html:27)
      File "/home/ubuntu/RootTheBox/models/Box.py", line 109, in all
        return dbsession.query(cls).all()
      File "/usr/local/lib/python2.7/dist-packages/sqlalchemy/orm/query.py", line 2843, in all
        return list(self)
      File "/usr/local/lib/python2.7/dist-packages/sqlalchemy/orm/query.py", line 2995, in __iter__
        return self._execute_and_instances(context)
      File "/usr/local/lib/python2.7/dist-packages/sqlalchemy/orm/query.py", line 3018, in _execute_and_instances
        result = conn.execute(querycontext.statement, self._params)
      File "/usr/local/lib/python2.7/dist-packages/sqlalchemy/engine/base.py", line 948, in execute
        return meth(self, multiparams, params)
      File "/usr/local/lib/python2.7/dist-packages/sqlalchemy/sql/elements.py", line 269, in _execute_on_connection
        return connection._execute_clauseelement(self, multiparams, params)
      File "/usr/local/lib/python2.7/dist-packages/sqlalchemy/engine/base.py", line 1060, in _execute_clauseelement
        compiled_sql, distilled_params
      File "/usr/local/lib/python2.7/dist-packages/sqlalchemy/engine/base.py", line 1200, in _execute_context
        context)
      File "/usr/local/lib/python2.7/dist-packages/sqlalchemy/engine/base.py", line 1413, in _handle_dbapi_exception
        exc_info
      File "/usr/local/lib/python2.7/dist-packages/sqlalchemy/util/compat.py", line 265, in raise_from_cause
        reraise(type(exception), exception, tb=exc_tb, cause=cause)
      File "/usr/local/lib/python2.7/dist-packages/sqlalchemy/engine/base.py", line 1193, in _execute_context
        context)
      File "/usr/local/lib/python2.7/dist-packages/sqlalchemy/engine/default.py", line 509, in do_execute
        cursor.execute(statement, parameters)
      File "/usr/local/lib/python2.7/dist-packages/MySQLdb/cursors.py", line 205, in execute
        self.errorhandler(self, exc, value)
      File "/usr/local/lib/python2.7/dist-packages/MySQLdb/connections.py", line 36, in defaulterrorhandler
        raise errorclass, errorvalue
    OperationalError: (_mysql_exceptions.OperationalError) (1054, "Unknown column 'box.flag_submission_type' in 'field list'") [SQL: u'SELECT box.id AS box_id, box.created AS box_created, box.uuid AS box_uuid, box.corporation_id AS box_corporation_id, box.category_id AS box_category_id, box._name AS box__name, box._operating_system AS box__operating_system, box._description AS box__description, box._difficulty AS box__difficulty, box.game_level_id AS box_game_level_id, box._avatar AS box__avatar, box.autoformat AS box_autoformat, box.garbage AS box_garbage, box.flag_submission_type AS box_flag_submission_type \nFROM box'] (Background on this error at: http://sqlalche.me/e/e3q8)

[E 181115 21:43:07 web:2162] 500 GET /admin/view/game_objects (x.x.x.x) 50.62ms

**Comment by jeffg2k** [comment]
Sorry, there was a version change since your last git update.  Run this SQL on your DB.

```
ALTER TABLE `rootthebox`.`box` 
ADD COLUMN `flag_submission_type` enum('CLASSIC','SINGLE_SUBMISSION_BOX') DEFAULT 'CLASSIC' AFTER `garbage`;
```

**Comment by mohammadHmedat** [comment]
if I put a URL in the box description, how to make it clickable?

**Comment by jeffg2k** [comment]
Don't support that at the moment. We'd have to implement some markup syntax, but not sure if that would work in the pre tags of the description box.

**Comment by jeffg2k** [comment]
I pushed a new branch that has [Markdown](https://guides.github.com/features/mastering-markdown/) support, which will allow you to add hyperlinks along with other formatting.  https://github.com/moloch--/RootTheBox/tree/Description-Markdown

While working on it, I rewrote the formatting of the description to remove "pre" tags, but maintaining a fixed width font.  This allowed some text justification, so that we don't have words cut off for new lines.  I think I got it looking very close to the original.  I also removed the autoformat option, since it's unnecessary with the justification pre-wrap.

@moloch-- you want to take a look at this one before I merge it in?  See if this would cause any security or formatting issues that I'm unaware of.
